src/endpoint.rs

```diff
@@ -361,27 +381,27 @@ mod tests {
 
     #[test]
     fn new_endpoints() {
-        assert!(Endpoints::new(vec![]).is_none());
-        assert!(Endpoints::new(vec![ep(1)]).is_some());
+        assert!(Endpoints::try_new(vec![]).is_none());
+        assert!(Endpoints::try_new(vec![ep(1)]).is_some());
```

Comment:
Since this PR also introduces the `new` constructor that could panic, would it make sense to add an explicit test for it so it's clear it's intentional and nothing temporary or bad practice? Or is it just too verbose and obvious?

```suggestion
        assert!(Endpoints::try_new(vec![]).is_none());
        assert!(Endpoints::try_new(vec![ep(1)]).is_some());
        assert_eq!(Endpoints::new(vec![ep(1)]).0.len(), 1);
    }
    
    #[test]
    #[should_panic]
    fn new_endpoints_panic() {
        Endpoints::new(vec![]);
```

Reply:
I've now added documentation about the panic, and a test for the behaviour.